### DISCUSSION TITLE
Request for self-starting of Ethernet sharing

### DISCUSSION THREAD

**Discussion by SatenRuiko-Lv0** [Mygod/VPNHotspot#632] [Q&A]
In the past year, I searched all available methods, such as shell instruction, adb instruction, magisk module, tasker and other automation software, but I couldn't find a way to open Ethernet sharing. Only your software can directly bypass the system settings and directly open the Ethernet share.Such an easy-to-use function is very suitable for Android phones as cpe to provide network connection to computers or routers, but it is a pity that you need to manually enter the settings to turn on this function every time you access Ethernet.
Is it possible to add automatic detection and turn on Ethernet sharing?Or is there any way you can directly open the function of Ethernet sharing through shell command?

**Comment by Mygod** [comment]
You can do this with Tasker or something similar which this app supports.

EDIT: I suppose only GitHub version has Tasker support yet. I will release a new version on Google Play later.

**Comment by sherylynn** [comment]
> In the past year, I searched all available methods, such as shell instruction, adb instruction, magisk module, tasker and other automation software, but I couldn't find a way to open Ethernet sharing. Only your software can directly bypass the system settings and directly open the Ethernet share.Such an easy-to-use function is very suitable for Android phones as cpe to provide network connection to computers or routers, but it is a pity that you need to manually enter the settings to turn on this function every time you access Ethernet. Is it possible to add automatic detection and turn on Ethernet sharing?Or is there any way you can directly open the function of Ethernet sharing through shell command?

you can try my version @SatenRuiko-Lv0   I had make PR

**Reply by andy2301** [reply]
@SatenRuiko-Lv0 not sure if you are still looking. You can try the following in Termux

> ##toggle eth tethering in Android Settings
> su -c input keyevent KEYCODE_WAKEUP # wake up the phone
> su -c input keyevent KEYCODE_HOME # go to home
> #su -c input keyevent KEYCODE_ENTER # enter sth
> su -c am start -n com.android.settings/.TetherSettings
> sleep 1
> su -c input keyevent KEYCODE_DPAD_DOWN # select the Wi-Fi hotspot row
> su -c input keyevent KEYCODE_DPAD_DOWN # move down to the next row (usb tethering)
> su -c input keyevent KEYCODE_DPAD_DOWN # move down to the next row (bluetooth tethering)
> su -c input keyevent KEYCODE_DPAD_DOWN # move down to the next row (ethernet tethering)
> su -c input keyevent KEYCODE_ENTER     # toggle ethernet tethering
